frontend/js/src/user/playlists/components/ImportPlaylistModal.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useContext, useState } from "react";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { toast } from "react-toastify";
+import GlobalAppContext from "../../../utils/GlobalAppContext";
+import { ToastMsg } from "../../../notifications/Notifications";
+
+export const MUSICBRAINZ_JSPF_PLAYLIST_EXTENSION =
+  "https://musicbrainz.org/doc/jspf#playlist";
+
+type ImportPLaylistModalProps = {
+  listenlist: JSPFTrack[];
+};
+
+export default NiceModal.create((props: ImportPLaylistModalProps) => {
+  const modal = useModal();
+
+  const closeModal = React.useCallback(() => {
+    modal.hide();
+    document?.body?.classList?.remove("modal-open");
+    setTimeout(modal.remove, 200);
+  }, [modal]);
+
+  const { APIService, currentUser } = React.useContext(GlobalAppContext);
+  const [playlists, setPlaylists] = React.useState<SpotifyPlaylistObject[]>([]);
+
+  React.useEffect(() => {
+    async function importPlaylistFromSpotify() {
+      try {
+        const response = await APIService.importPlaylistToSpotify(
+          currentUser?.auth_token
+        );
+
+        if (!response) {
+          return;
+        }
+
+        setPlaylists(response);
+      } catch (error) {
+        toast.error(
+          <ToastMsg
+            title="Error loading playlists"
+            message={error?.message ?? error}
+          />,
+          { toastId: "load-playlists-error" }
+        );
+      }
+    }
+
+    if (currentUser?.auth_token) {
+      importPlaylistFromSpotify();
+    }
+  }, [APIService, currentUser]);
+
+  const alertMustBeLoggedIn = () => {
+    toast.error(
+      <ToastMsg
+        title="Error"
+        message="You must be logged in for this operation"
+      />,
+      { toastId: "auth-error" }
+    );
+  };
+
+  const importTracksToPlaylist = async (
+    playlistID: string,
+    playlistName: string
+  ) => {
+    if (!currentUser?.auth_token) {
+      alertMustBeLoggedIn();
+      return;
+    }
+
+    try {
```

Comment:
One thing we are missing here is a loading indicator,
Since the import can take some time, it would be useful for users to see that something is going on after they clicked on a playlist name.

You can store a list of playlist IDs in state (say `playlistsSelectedForImport` or something).
Then you could show an indicator directly on the item in the list of playlists, which will make it clear after clicking that import is in progress.

After a successful or failed playlist import, don't forget to remove said playlist ID from the (playlistsSelectedForImport) state to remove the indicator.